### PR TITLE
Fix the dir/baseDir property setting in the FileCache file

### PR DESCRIPTION
**Pull request by coudenysj**
Change the order of the directory creation and property setting so the property holds the correct value

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
@lsmith77 that seems to be a fix to your fix, right? And i think it should be done on 1.1 and master, right?

File: Cache/FileCache.php [hunk 1]
```diff
@@ -29,20 +29,20 @@ class FileCache implements CacheInterface, CacheWarmerInterface, CacheClearerInt
...
+            throw new \InvalidArgumentException(sprintf('The directory "%s" is not writable.', $baseDir));
         }
 
-        $this->dir = rtrim($dir, '\\/');
-
-        $this->umask = $umask;
-
         $dir = $baseDir.DIRECTORY_SEPARATOR.$dir;
         if (!is_dir($dir)) {
             mkdir($dir, 0777 & ~$this->umask, true);
```
Comment (ElectricMaxxx) [inline Cache/FileCache.php]: `$this->unmask` does not exist at, now.

Comment (ElectricMaxxx) [inline Cache/FileCache.php]: I think you should use `$unmask`instad, or keep the order for the `$this->unmask = $unmaks`

Comment (coudenysj) [inline Cache/FileCache.php]: Correct, let me fine tune a bit

**Comment by lsmith77** [comment]
its not really a fix to my fix .. it just fixes another issue in the same code block .. but yes this should also be put into 1.1

**Comment by coudenysj** [comment]
It fixes the changed line order in 3b8b21c795982df4895bda4896003340a1226a8b.

**Comment by lsmith77** [comment]
ah ok ..

BTW it would be good to add a test for this ..

**Pull request merged**